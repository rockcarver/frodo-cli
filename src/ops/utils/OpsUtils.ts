import { FrodoError } from '@rockcarver/frodo-lib';

import { printError } from '../../utils/Console';

/**
 * Deep clone object
 * @param {any} obj object to deep clone
 * @returns {any} new object cloned from obj
 */
export function cloneDeep(obj: any): any {
  return JSON.parse(JSON.stringify(obj));
}

/**
 * Result callback that handles errors when they occur by printing them and setting the process exit code
 * @param error The error to handle
 */
export function errorHandler(error?: FrodoError) {
  if (!error) return;
  printError(error);
  process.exitCode = 1;
}
