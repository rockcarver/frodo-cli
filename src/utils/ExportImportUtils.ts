import fs from 'fs';
import slugify from 'slugify';
import { ExportImportUtils, state } from '@rockcarver/frodo-lib';
import { printMessage } from './Console';

const { getMetadata } = ExportImportUtils;

/**
 * Get a typed filename. E.g. "my-script.script.json"
 *
 * @param name The name of the file
 * @param type The type of the file, e.g. script, idp, etc.
 * @param suffix The suffix of the file, e.g. json, xml, etc. Defaults to json.
 * @returns The typed filename
 */
export function getTypedFilename(
  name: string,
  type: string,
  suffix = 'json'
): string {
  const slug = slugify(name.replace(/^http(s?):\/\//, ''));
  return `${slug}.${type}.${suffix}`;
}

/**
 * Save JSON object to file
 *
 * @param data data object
 * @param filename file name
 */
export function saveJsonToFile(data: any, filename: string) {
  const exportData = data;
  if (!exportData.meta) {
    exportData.meta = getMetadata();
  }
  fs.writeFile(filename, JSON.stringify(exportData, null, 2), (err) => {
    if (err) {
      return printMessage(`ERROR - can't save ${filename}`, 'error');
    }
    return '';
  });
}

export function saveToFile(type, data, identifier, filename) {
  const exportData = {};
  exportData['meta'] = getMetadata();
  exportData[type] = {};

  if (Array.isArray(data)) {
    data.forEach((element) => {
      exportData[type][element[identifier]] = element;
    });
  } else {
    exportData[type][data[identifier]] = data;
  }
  fs.writeFile(filename, JSON.stringify(exportData, null, 2), (err) => {
    if (err) {
      return printMessage(`ERROR - can't save ${type} to file`, 'error');
    }
    return '';
  });
}

/**
 * Save text data to file
 * @param data text data
 * @param filename file name
 * @return true if successful, false otherwise
 */
export function saveTextToFile(data: string, filename: string): boolean {
  try {
    fs.writeFileSync(filename, data);
    return true;
  } catch (error) {
    printMessage(`ERROR - can't save ${filename}`, 'error');
    return false;
  }
}

/*
 * Output str in title case
 *
 * e.g.: 'ALL UPPERCASE AND all lowercase' = 'All Uppercase And All Lowercase'
 */
export function titleCase(input) {
  const str = input.toString();
  const splitStr = str.toLowerCase().split(' ');
  for (let i = 0; i < splitStr.length; i += 1) {
    splitStr[i] = splitStr[i].charAt(0).toUpperCase() + splitStr[i].slice(1);
  }
  return splitStr.join(' ');
}

export function getRealmString() {
  const realm = state.getRealm();
  return realm
    .split('/')
    .reduce((result, item) => `${result}${titleCase(item)}`, '');
}
