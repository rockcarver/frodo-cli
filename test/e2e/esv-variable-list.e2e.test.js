/**
 * Follow this process to write e2e tests for the CLI project:
 *
 * 1. Test if all the necessary mocks for your tests already exist.
 *    In mock mode, run the command you want to test with the same arguments
 *    and parameters exactly as you want to test it, for example:
 *
 *    $ FRODO_MOCK=1 frodo conn save https://openam-frodo-dev.forgeblocks.com/am volker.scheuber@forgerock.com Sup3rS3cr3t!
 *
 *    If your command completes without errors and with the expected results,
 *    all the required mocks already exist and you are good to write your
 *    test and skip to step #4.
 *
 *    If, however, your command fails and you see errors like the one below,
 *    you know you need to record the mock responses first:
 *
 *    [Polly] [adapter:node-http] Recording for the following request is not found and `recordIfMissing` is `false`.
 *
 * 2. Record mock responses for your exact command.
 *    In mock record mode, run the command you want to test with the same arguments
 *    and parameters exactly as you want to test it, for example:
 *
 *    $ FRODO_MOCK=record frodo conn save https://openam-frodo-dev.forgeblocks.com/am volker.scheuber@forgerock.com Sup3rS3cr3t!
 *
 *    Wait until you see all the Polly instances (mock recording adapters) have
 *    shutdown before you try to run step #1 again.
 *    Messages like these indicate mock recording adapters shutting down:
 *
 *    Polly instance 'conn/4' stopping in 3s...
 *    Polly instance 'conn/4' stopping in 2s...
 *    Polly instance 'conn/save/3' stopping in 3s...
 *    Polly instance 'conn/4' stopping in 1s...
 *    Polly instance 'conn/save/3' stopping in 2s...
 *    Polly instance 'conn/4' stopped.
 *    Polly instance 'conn/save/3' stopping in 1s...
 *    Polly instance 'conn/save/3' stopped.
 *
 * 3. Validate your freshly recorded mock responses are complete and working.
 *    Re-run the exact command you want to test in mock mode (see step #1).
 *
 * 4. Write your test.
 *    Make sure to use the exact command including number of arguments and params.
 *
 * 5. Commit both your test and your new recordings to the repository.
 *    Your tests are likely going to reside outside the frodo-lib project but
 *    the recordings must be committed to the frodo-lib project.
 */

/*
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo esv variable list
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo esv variable list -l
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo esv variable list -u
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo esv variable list -lu
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo esv variable list -uf test/e2e/exports/all/all.cloud.json
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo esv variable list --usage --long --file test/e2e/exports/all/all.cloud.json
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo esv variable list -uD test/e2e/exports/all-separate/cloud
FRODO_MOCK=record FRODO_NO_CACHE=1 FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am frodo esv variable list --usage --long --directory test/e2e/exports/all-separate/cloud
 */
import cp from 'child_process';
import { promisify } from 'util';
import { getEnv, removeAnsiEscapeCodes, removeProgressBarOutput } from './utils/TestUtils';
import { connection as c } from './utils/TestConfig';

const exec = promisify(cp.exec);

process.env['FRODO_MOCK'] = '1';
const env = getEnv(c);

const allConfigFile = 'test/e2e/exports/all/all.cloud.json';
const allConfigDirectory = 'test/e2e/exports/all-separate/cloud';

describe('frodo esv variable list', () => {
    test('"frodo esv variable list": should list the ids of the esv variables', async () => {
        const CMD = `frodo esv variable list`;
        const { stdout } = await exec(CMD, env);
        expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
    });

    test('"frodo esv variable list -l": should list the ids, values, statuses, descriptions, modifiers, and modified times of the esv variables', async () => {
        const CMD = `frodo esv variable list -l`;
        const { stdout } = await exec(CMD, env);
        expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
    });

    test('"frodo esv variable list -u": should list the usage of the esv variables', async () => {
        const CMD = `frodo esv variable list -u`;
        try {
            await exec(CMD, env);
            fail("Command should've failed")
        } catch (e) {
            expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
            expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stdout))).toMatchSnapshot();
        }
    });

    test('"frodo esv variable list -lu": should list the ids, values, statuses, descriptions, modifiers, usage, and modified times of the esv variables', async () => {
        const CMD = `frodo esv variable list -lu`;
        try {
            await exec(CMD, env);
            fail("Command should've failed")
        } catch (e) {
            expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stderr))).toMatchSnapshot();
            expect(removeProgressBarOutput(removeAnsiEscapeCodes(e.stdout))).toMatchSnapshot();
        }
    });

    test(`"frodo esv variable list -uf ${allConfigFile}": should list the usage of the esv variables in the ${allConfigFile} file`, async () => {
        const CMD = `frodo esv variable list -uf ${allConfigFile}`;
        const { stdout } = await exec(CMD, env);
        expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
    });

    test(`"frodo esv variable list --usage --long --file ${allConfigFile}": should list the ids, values, statuses, descriptions, modifiers, usage, and modified times of the esv variables in the ${allConfigFile} file`, async () => {
        const CMD = `frodo esv variable list --usage --long --file ${allConfigFile}`;
        const { stdout } = await exec(CMD, env);
        expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
    });

    test(`"frodo esv variable list -uD ${allConfigDirectory}": should list the usage of the esv variables in the ${allConfigDirectory} directory`, async () => {
        const CMD = `frodo esv variable list -uD ${allConfigDirectory}`;
        const { stdout } = await exec(CMD, env);
        expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
    });

    test(`"frodo esv variable list --usage --long --directory ${allConfigDirectory}": should list the ids, values, statuses, descriptions, modifiers, usage, and modified times of the esv variables in the ${allConfigDirectory} directory`, async () => {
        const CMD = `frodo esv variable list --usage --long --directory ${allConfigDirectory}`;
        const { stdout } = await exec(CMD, env);
        expect(removeAnsiEscapeCodes(stdout)).toMatchSnapshot();
    });
});
