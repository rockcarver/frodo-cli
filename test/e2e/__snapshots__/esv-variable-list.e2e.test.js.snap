// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`frodo esv variable list "frodo esv variable list --usage --long --directory test/e2e/exports/all-separate/cloud": should list the ids, values, statuses, descriptions, modifiers, usage, and modified times of the esv variables in the test/e2e/exports/all-separate/cloud directory 1`] = `
"Id                               │Value                                   │Status│Description                             │Modifier                     │Modified (UTC)               │Used                                                                                                                                          
esv-blue-piller                  │false                                   │loaded│Zion membership criteria.               │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:11 GMT│no                                                                                                                                            
esv-ipv4-cidr-access-rules       │{ "allow": [ "145.118.0.0/16",          │loaded│IPv4 CIDR access rules: { "allow": [    │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:13 GMT│no                                                                                                                                            
                                 │"132.35.0.0/16", "101.226.0.0/16",      │      │"address/mask" ] }                      │                             │                             │                                                                                                                                              
                                 │"99.72.28.182/32" ] }                   │      │                                        │                             │                             │                                                                                                                                              
esv-nebuchadnezzar-crew          │,"Apoc","Cypher","Mouse","Neo","Switch"]│loaded│The crew of the Nebuchadnezzar          │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:05 GMT│no                                                                                                                                            
                                 │                                        │      │hovercraft.                             │                             │                             │                                                                                                                                              
esv-nebuchadnezzar-crew-structure│"Apoc","Cypher","Mouse","Neo","Switch"]}│loaded│The structure of the crew of the        │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:07 GMT│no                                                                                                                                            
                                 │                                        │      │Nebuchadnezzar hovercraft.              │                             │                             │                                                                                                                                              
esv-neo-age                      │28                                      │loaded│Neo's age in the matrix.                │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:09 GMT│no                                                                                                                                            
esv-number                       │1.134                                   │loaded│test number                             │volker.scheuber@forgerock.com│Fri, 05 Jul 2024 19:42:20 GMT│no                                                                                                                                            
esv-test-var                     │this is a test variable                 │loaded│this is a test description              │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 22:37:25 GMT│no                                                                                                                                            
esv-test-var-pi                  │3.1415926                               │loaded│This is another test variable.          │Frodo-SA-1720799681233       │Fri, 12 Jul 2024 17:40:41 GMT│yes (2 uses, including: realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script.0)
esv-test-var-pi-string           │3.1415926                               │loaded│This is another test variable.          │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:16 GMT│no                                                                                                                                            
esv-test-variable-light          │299792458                               │loaded│Test variable containing the speed of   │phales@trivir.com            │Thu, 14 Dec 2023 15:34:13 GMT│yes (at realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script.2)                
                                 │                                        │      │light in meters per second (as an int). │                             │                             │                                                                                                                                              
esv-trinity-phone                │(312)-555-0690                          │loaded│In the opening of The Matrix (1999), the│Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:03 GMT│no                                                                                                                                            
                                 │                                        │      │phone number Trinity is calling from is │                             │                             │                                                                                                                                              
                                 │                                        │      │traced to (312)-555-0690                │                             │                             │                                                                                                                                              
"
`;

exports[`frodo esv variable list "frodo esv variable list --usage --long --file test/e2e/exports/all/all.cloud.json": should list the ids, values, statuses, descriptions, modifiers, usage, and modified times of the esv variables in the test/e2e/exports/all/all.cloud.json file 1`] = `
"Id                               │Value                                   │Status│Description                             │Modifier                     │Modified (UTC)               │Used                                                                                                                        
esv-blue-piller                  │false                                   │loaded│Zion membership criteria.               │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:11 GMT│no                                                                                                                          
esv-ipv4-cidr-access-rules       │{ "allow": [ "145.118.0.0/16",          │loaded│IPv4 CIDR access rules: { "allow": [    │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:13 GMT│no                                                                                                                          
                                 │"132.35.0.0/16", "101.226.0.0/16",      │      │"address/mask" ] }                      │                             │                             │                                                                                                                            
                                 │"99.72.28.182/32" ] }                   │      │                                        │                             │                             │                                                                                                                            
esv-nebuchadnezzar-crew          │,"Apoc","Cypher","Mouse","Neo","Switch"]│loaded│The crew of the Nebuchadnezzar          │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:05 GMT│no                                                                                                                          
                                 │                                        │      │hovercraft.                             │                             │                             │                                                                                                                            
esv-nebuchadnezzar-crew-structure│"Apoc","Cypher","Mouse","Neo","Switch"]}│loaded│The structure of the crew of the        │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:07 GMT│no                                                                                                                          
                                 │                                        │      │Nebuchadnezzar hovercraft.              │                             │                             │                                                                                                                            
esv-neo-age                      │28                                      │loaded│Neo's age in the matrix.                │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:09 GMT│no                                                                                                                          
esv-number                       │1.134                                   │loaded│test number                             │volker.scheuber@forgerock.com│Fri, 05 Jul 2024 19:42:20 GMT│no                                                                                                                          
esv-test-var                     │this is a test variable                 │loaded│this is a test description              │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 22:37:25 GMT│no                                                                                                                          
esv-test-var-pi                  │3.1415926                               │loaded│This is another test variable.          │Frodo-SA-1720799681233       │Fri, 12 Jul 2024 17:40:41 GMT│yes (at realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script)
esv-test-var-pi-string           │3.1415926                               │loaded│This is another test variable.          │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:16 GMT│no                                                                                                                          
esv-test-variable-light          │299792458                               │loaded│Test variable containing the speed of   │phales@trivir.com            │Thu, 14 Dec 2023 15:34:13 GMT│yes (at realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script)
                                 │                                        │      │light in meters per second (as an int). │                             │                             │                                                                                                                            
esv-trinity-phone                │(312)-555-0690                          │loaded│In the opening of The Matrix (1999), the│Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:03 GMT│no                                                                                                                          
                                 │                                        │      │phone number Trinity is calling from is │                             │                             │                                                                                                                            
                                 │                                        │      │traced to (312)-555-0690                │                             │                             │                                                                                                                            
"
`;

exports[`frodo esv variable list "frodo esv variable list -l": should list the ids, values, statuses, descriptions, modifiers, and modified times of the esv variables 1`] = `
"Id                                          │Value                          │Status│Description                            │Modifier                                      │Modified (UTC)               
esv-global-session-quotas-enable-constraints│ON                             │loaded│                                       │sandeep.chaturvedi@forgerock.com              │Mon, 22 Jan 2024 16:35:58 GMT
esv-test-var                                │true                           │loaded│Simple ESV test variable with a boolean│unknown (76618ff6-e851-433e-9704-9d2852a17b7a)│Mon, 18 Dec 2023 23:02:35 GMT
                                            │                               │      │value of true                          │                                              │                             
esv-test-variable-belphegor                 │1000000000000066600000000000001│loaded│A test variable containing belphegor's │phales@trivir.com                             │Thu, 14 Dec 2023 15:31:06 GMT
                                            │                               │      │palindromic prime number (as a string).│                                              │                             
esv-test-variable-light                     │299792458                      │loaded│Test variable containing the speed of  │phales@trivir.com                             │Thu, 14 Dec 2023 15:34:13 GMT
                                            │                               │      │light in meters per second (as an int).│                                              │                             
"
`;

exports[`frodo esv variable list "frodo esv variable list -lu": should list the ids, values, statuses, descriptions, modifiers, usage, and modified times of the esv variables 1`] = `
"Connected to https://openam-frodo-dev.forgeblocks.com/am [alpha] as service account Frodo-SA-1720799681233 [b672336b-41ef-428d-ae4a-e0c082875377]
- Reading variables...
✔ Successfully read 12 variables.
Error exporting idm config entity fidc/federation-EntraID
  Error reading config entity fidc/federation-EntraID
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 403
    Reason: Forbidden
    Message: Access denied
"
`;

exports[`frodo esv variable list "frodo esv variable list -lu": should list the ids, values, statuses, descriptions, modifiers, usage, and modified times of the esv variables 2`] = `
"Id                               │Value                                   │Status│Description                             │Modifier                     │Modified (UTC)               │Used                                                                                                                                          
esv-blue-piller                  │false                                   │loaded│Zion membership criteria.               │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:11 GMT│no                                                                                                                                            
esv-ipv4-cidr-access-rules       │{ "allow": [ "145.118.0.0/16",          │loaded│IPv4 CIDR access rules: { "allow": [    │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:13 GMT│no                                                                                                                                            
                                 │"132.35.0.0/16", "101.226.0.0/16",      │      │"address/mask" ] }                      │                             │                             │                                                                                                                                              
                                 │"99.72.28.182/32" ] }                   │      │                                        │                             │                             │                                                                                                                                              
esv-nebuchadnezzar-crew          │,"Apoc","Cypher","Mouse","Neo","Switch"]│loaded│The crew of the Nebuchadnezzar          │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:05 GMT│no                                                                                                                                            
                                 │                                        │      │hovercraft.                             │                             │                             │                                                                                                                                              
esv-nebuchadnezzar-crew-structure│"Apoc","Cypher","Mouse","Neo","Switch"]}│loaded│The structure of the crew of the        │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:07 GMT│no                                                                                                                                            
                                 │                                        │      │Nebuchadnezzar hovercraft.              │                             │                             │                                                                                                                                              
esv-neo-age                      │28                                      │loaded│Neo's age in the matrix.                │Frodo-SA-1730238488278       │Fri, 01 Nov 2024 16:21:14 GMT│no                                                                                                                                            
esv-number                       │1.134                                   │loaded│test number                             │volker.scheuber@forgerock.com│Fri, 05 Jul 2024 19:42:20 GMT│no                                                                                                                                            
esv-test                         │a,b,c,d                                 │loaded│list                                    │phales@trivir.com            │Fri, 01 Nov 2024 21:00:21 GMT│no                                                                                                                                            
esv-test-var                     │this is a test variable                 │loaded│this is a test description              │Frodo-SA-1730238488278       │Fri, 01 Nov 2024 16:21:15 GMT│no                                                                                                                                            
esv-test-var-pi                  │3.1415926                               │loaded│This is another test variable.          │Frodo-SA-1720799681233       │Fri, 12 Jul 2024 17:40:41 GMT│yes (2 uses, including: realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script.0)
esv-test-var-pi-string           │3.1415926                               │loaded│This is another test variable.          │Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:16 GMT│no                                                                                                                                            
esv-test-variable-light          │299792458                               │loaded│Test variable containing the speed of   │phales@trivir.com            │Thu, 14 Dec 2023 15:34:13 GMT│yes (at realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script.2)                
                                 │                                        │      │light in meters per second (as an int). │                             │                             │                                                                                                                                              
esv-trinity-phone                │(312)-555-0690                          │loaded│In the opening of The Matrix (1999), the│Frodo-SA-1701393386423       │Fri, 05 Jul 2024 20:01:03 GMT│no                                                                                                                                            
                                 │                                        │      │phone number Trinity is calling from is │                             │                             │                                                                                                                                              
                                 │                                        │      │traced to (312)-555-0690                │                             │                             │                                                                                                                                              
"
`;

exports[`frodo esv variable list "frodo esv variable list -u": should list the usage of the esv variables 1`] = `
"Connected to https://openam-frodo-dev.forgeblocks.com/am [alpha] as service account Frodo-SA-1720799681233 [b672336b-41ef-428d-ae4a-e0c082875377]
- Reading variables...
✔ Successfully read 12 variables.
Error exporting idm config entity fidc/federation-EntraID
  Error reading config entity fidc/federation-EntraID
  HTTP client error
    Code: ERR_BAD_REQUEST
    Status: 403
    Reason: Forbidden
    Message: Access denied
"
`;

exports[`frodo esv variable list "frodo esv variable list -u": should list the usage of the esv variables 2`] = `
"Id                               │Used                                                                                                                                          
esv-blue-piller                  │no                                                                                                                                            
esv-ipv4-cidr-access-rules       │no                                                                                                                                            
esv-nebuchadnezzar-crew          │no                                                                                                                                            
esv-nebuchadnezzar-crew-structure│no                                                                                                                                            
esv-neo-age                      │no                                                                                                                                            
esv-number                       │no                                                                                                                                            
esv-test                         │no                                                                                                                                            
esv-test-var                     │no                                                                                                                                            
esv-test-var-pi                  │yes (2 uses, including: realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script.0)
esv-test-var-pi-string           │no                                                                                                                                            
esv-test-variable-light          │yes (at realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script.2)                
esv-trinity-phone                │no                                                                                                                                            
"
`;

exports[`frodo esv variable list "frodo esv variable list -uD test/e2e/exports/all-separate/cloud": should list the usage of the esv variables in the test/e2e/exports/all-separate/cloud directory 1`] = `
"Id                               │Used                                                                                                                                          
esv-blue-piller                  │no                                                                                                                                            
esv-ipv4-cidr-access-rules       │no                                                                                                                                            
esv-nebuchadnezzar-crew          │no                                                                                                                                            
esv-nebuchadnezzar-crew-structure│no                                                                                                                                            
esv-neo-age                      │no                                                                                                                                            
esv-number                       │no                                                                                                                                            
esv-test-var                     │no                                                                                                                                            
esv-test-var-pi                  │yes (2 uses, including: realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script.0)
esv-test-var-pi-string           │no                                                                                                                                            
esv-test-variable-light          │yes (at realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script.2)                
esv-trinity-phone                │no                                                                                                                                            
"
`;

exports[`frodo esv variable list "frodo esv variable list -uf test/e2e/exports/all/all.cloud.json": should list the usage of the esv variables in the test/e2e/exports/all/all.cloud.json file 1`] = `
"Id                               │Used                                                                                                                        
esv-blue-piller                  │no                                                                                                                          
esv-ipv4-cidr-access-rules       │no                                                                                                                          
esv-nebuchadnezzar-crew          │no                                                                                                                          
esv-nebuchadnezzar-crew-structure│no                                                                                                                          
esv-neo-age                      │no                                                                                                                          
esv-number                       │no                                                                                                                          
esv-test-var                     │no                                                                                                                          
esv-test-var-pi                  │yes (at realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script)
esv-test-var-pi-string           │no                                                                                                                          
esv-test-variable-light          │yes (at realm.root-alpha.script.da7a96a8-7969-4dab-9c6e-a812938cc76d(name: 'test-script-with-secrets-and-variables').script)
esv-trinity-phone                │no                                                                                                                          
"
`;

exports[`frodo esv variable list "frodo esv variable list": should list the ids of the esv variables 1`] = `
"esv-global-session-quotas-enable-constraints
esv-test-var
esv-test-variable-belphegor
esv-test-variable-light
"
`;
